From 283aa25f757053d99515887348cdc49e873e577c Mon Sep 17 00:00:00 2001
From: David Hildenbrand <david@redhat.com>
Date: Mon, 8 Jan 2018 21:26:31 +0100
Subject: [PATCH 01/12] tools/kvm_stat: fix command line option '-g'

RH-Author: David Hildenbrand <david@redhat.com>
Message-id: <20180108212631.29046-1-david@redhat.com>
Patchwork-id: 78528
O-Subject: [RHEL-7.5 qemu-kvm-rhev PATCH] tools/kvm_stat: fix command line option '-g'
Bugzilla: 1529676
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>

commit 19e8e54f4309eaa438237aa1973fe40c331903d4 (HEAD)
Author: Stefan Raspl <stefan.raspl@de.ibm.com>
Date:   Mon Dec 11 12:25:19 2017 +0100

    tools/kvm_stat: fix command line option '-g'

    Specifying a guest via '-g foo' always results in an error:
      $ kvm_stat -g foo
      Usage: kvm_stat [options]

      kvm_stat: error: Error while searching for guest "foo", use "-p" to
      specify a pid instead

    Reason is that Tui.get_pid_from_gname() is not static, as it is supposed
    to be.

    Signed-off-by: Stefan Raspl <raspl@linux.vnet.ibm.com>
    Tested-by: Christian Borntraeger <borntraeger@de.ibm.com>
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

Signed-off-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 scripts/kvm/kvm_stat | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/scripts/kvm/kvm_stat b/scripts/kvm/kvm_stat
index 32283d8..c74a9a0 100755
--- a/scripts/kvm/kvm_stat
+++ b/scripts/kvm/kvm_stat
@@ -946,7 +946,8 @@ class Tui(object):
             curses.nocbreak()
             curses.endwin()
 
-    def get_all_gnames(self):
+    @staticmethod
+    def get_all_gnames():
         """Returns a list of (pid, gname) tuples of all running guests"""
         res = []
         try:
@@ -959,7 +960,7 @@ class Tui(object):
             # perform a sanity check before calling the more expensive
             # function to possibly extract the guest name
             if ' -name ' in line[1]:
-                res.append((line[0], self.get_gname_from_pid(line[0])))
+                res.append((line[0], Tui.get_gname_from_pid(line[0])))
         child.stdout.close()
 
         return res
@@ -980,7 +981,8 @@ class Tui(object):
         except Exception:
             self.screen.addstr(row + 1, 2, 'Not available')
 
-    def get_pid_from_gname(self, gname):
+    @staticmethod
+    def get_pid_from_gname(gname):
         """Fuzzy function to convert guest name to QEMU process pid.
 
         Returns a list of potential pids, can be empty if no match found.
@@ -988,7 +990,7 @@ class Tui(object):
 
         """
         pids = []
-        for line in self.get_all_gnames():
+        for line in Tui.get_all_gnames():
             if gname == line[1]:
                 pids.append(int(line[0]))
 
-- 
1.8.3.1

